src/Nokia_LCD.cpp

```diff
@@ -248,7 +258,27 @@ void Nokia_LCD::sendCommand(const unsigned char command) {
 
 bool Nokia_LCD::sendData(const unsigned char data) { return send(data, true); }
 
-bool Nokia_LCD::send(const unsigned char lcd_byte, const bool is_data) {
+
+bool Nokia_LCD::updateCursorPosition(const unsigned int x_start_position, const unsigned int x_end_position) {
+    bool out_of_bounds = false;
+
+    mX_cursor = (mX_cursor + 1) % (x_end_position + x_start_position);  // Column
+    // Calculate the cursor position after the byte being sent
+    if (mX_cursor == 0) {
+        mX_cursor = x_start_position;      // return X to the initial position 
+        // If the column just became 0, this means the row should change
+        mY_cursor = (mY_cursor + 1) % kTotal_rows;  // Row
+        if (mY_cursor == 0) {
+            // If we are back to row 0 again, then we just went out of bounds
+            out_of_bounds = true;
+        }
+    }
+
+    setCursor(mX_cursor, mY_cursor);     // update the cursor position   
+    return out_of_bounds;
+}
+
+bool Nokia_LCD::send(const unsigned char lcd_byte, const bool is_data, const bool update_cursor) {
```

Comment:
I could add the param `update_cursor` in the sendData( ) method too, so that I can call sendData instead of send() in draw() method.  What do you think?

Reply:
Is there a reason someone would want to update the cursor when sending data/commands to the screen?

Reply:
I believe only for sending data, commands do not. I will change only the sendData method and add the param optional too.

Reply:
Why would someone want to send data to be presented on the screen and not update the cursor?
I don't understand why the second argument needs to be there.

Reply:
OK, after looking at the code more carefully, I see that you need the second argument for **internal usage**, i.e. for doing some custom update of the cursor position when drawing images that are not the "default" width.

The user should not care (even if it's a default argument) about whether they need to update the cursor position or not, because they want to send data that are presented on the screen.

I suggest the following:
Leave the **public** `sendData` as it was and create a **private** overload that takes the `update_cursor` argument (no defaults in the private method). The **public** method calls the overloaded **private** one with `true` as the second argument and  **internally**, we call the private method with the `update_cursor` argument appropriately set to `false`.
This way we don't expose internal design choices to the user who should not care about whether they need to update the cursor or not. From their perspective the cursor position should always be auto-magically updated when they send data to be presented on the screen.

Reply:
Cool, I got it. Pushing changes.